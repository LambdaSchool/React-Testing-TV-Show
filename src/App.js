import React, { useState, useEffect } from "react";
<<<<<<< HEAD
// import axios from "axios";
=======
>>>>>>> 0c5d3ee399b31009f03fdad8b8318a02990ec8c7
import Dropdown from "react-dropdown";
import parse from "html-react-parser";

import { formatSeasons } from "./utils/formatSeasons";
import { fetchShow } from "./api/fetchShow";

import Episodes from "./components/Episodes";
import "./styles.css";
import { fetchShow } from "./api/fetchShow";

export default function App() {
  const [show, setShow] = useState(null);
  const [seasons, setSeasons] = useState([]);
  const [selectedSeason, setSelectedSeason] = useState("");
  const episodes = seasons[selectedSeason] || [];

  useEffect(() => {
    fetchShow().then(res => {
    fetchShow()
    .then(res => {
      console.log(res)
      setShow(res);
      setSeasons(formatSeasons(res._embedded.episodes));
    })
    .catch(err => console.log(err))
  }, []);

  const handleSelect = (e) => {
    setSelectedSeason(e.value);
  };

  if (!show) {
    return <h2>Loading data...</h2>;
  }
  return (
    <div className="App">
      <img className="poster-img" src={show.image.original} alt={show.name} />
      <h1>{show.name}</h1>
      {parse(show.summary)}
      <Dropdown
        options={Object.keys(seasons)}
        onChange={handleSelect}
        value={selectedSeason || "Select a season"}
        placeholder="Select an option"
        data-testid="dropdown"
      />
      <Episodes episodes={episodes} />
    </div>
  );
}