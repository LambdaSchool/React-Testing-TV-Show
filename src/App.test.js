import App from "./App";
import {render} from "@testing-library/react";

// Sample Data 

const show = {   	
id: 2993,
url: "http://www.tvmaze.com/shows/2993/stranger-things",
name: "Stranger Things",
type: "Scripted",
language: "English",
genres: {	
0: "Drama",
1: "Fantasy",
2: "Science-Fiction",
},
status: "Running",
runtime: 60,
premiered: "2016-07-15",
officialSite: "https://www.netflix.com/title/80057281",
}


test('the imported data is correct', () => {
    expect(App).toMatchSnapshot();
});

// Mocking test 
test('mock implementation of a basic function', () => {
    const mock = jest.fn(() => "I am a mock function");
    console.log(mock);
    expect(mock('Calling my mock function!')).toBe("I am a mock function");
    expect(mock).toHaveBeenCalledWith('Calling my mock function!');
});

test("Runs and renders App", () => {
    const test = render(<App />)
    screen.debug()
})

