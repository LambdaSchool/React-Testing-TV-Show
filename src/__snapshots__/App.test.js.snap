// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the imported data is correct 1`] = `[Function]`;
